lib/IDE/CodeCompletion.cpp

```diff
@@ -1371,12 +1599,21 @@ MutableArrayRef<CodeCompletionResult *> CodeCompletionContext::takeResults() {
 Optional<unsigned> CodeCompletionString::getFirstTextChunkIndex(
     bool includeLeadingPunctuation) const {
   for (auto i : indices(getChunks())) {
-    auto &C = getChunks()[i];
+    const Chunk &C = getChunks()[i];
     switch (C.getKind()) {
     using ChunkKind = Chunk::ChunkKind;
     case ChunkKind::Text:
+      // Skip white-space only chunks.
+      if (C.getText().find_first_not_of(" \r\n") == StringRef::npos)
+        continue;
+      return i;
     case ChunkKind::CallArgumentName:
     case ChunkKind::CallArgumentInternalName:
+    case ChunkKind::ParameterDeclExternalName:
+    case ChunkKind::ParameterDeclLocalName:
+    case ChunkKind::ParameterDeclColon:
+    case ChunkKind::GenericParameterClauseBegin:
+    case ChunkKind::GenericRequirementClauseBegin:
```

Comment:
Don't these `Begin` cases return `false` from `chunkHasText()`?

Reply:
These chunks themselves don't have text, but we should consider these *groups* have text. Let's say we have something like:
```
- (Whitespace " ")
- | (ClauseBegin)
- | (Text "something")
- | (Text "other")
- (Text "outside")
```
We want to "print" `ClauseBegin` chunk as well because its group has some text.